contracts/HyperVIBES.sol

```diff
@@ -405,6 +443,30 @@ contract HyperVIBES {
         emit Claimed(input.realmId, input.collection, input.tokenId, toClaim);
     }
 
+    // returns true if msg.sender can claim for a given (realm/collection/tokenId) tuple
+    function _isValidClaimer(uint256 realmId, IERC721 collection, uint256 tokenId) internal view returns (bool) {
```

Comment:
the cyclomatic complexity of this function is horrendous but this seems mostly readable

* **IF** msg.sender is NFT owner **OR** is approved / approved for all
  * **IF** public claiming is allowed for the realm -> **return true**
  * **IF** msg.sender is on the claimer allow list -> **return true**
* **ELSE** return false